common/types/activation.go

```diff
@@ -299,18 +241,28 @@ func (atx *ActivationTx) MarshalLogObject(encoder log.ObjectEncoder) error {
 	return nil
 }
 
-// CalcAndSetID calculates and sets the cached ID field. This field must be set before calling the ID() method.
-func (atx *ActivationTx) CalcAndSetID() error {
+// Initialize calculates and sets the cached ID field. This field must be set before calling the ID() method.
+func (atx *ActivationTx) Initialize() error {
+	if atx.ID() != EmptyATXID {
+		return fmt.Errorf("ATX already initialized")
+	}
+
 	if atx.Signature == EmptyEdSignature {
-		return fmt.Errorf("cannot calculate ATX ID: sig is nil")
+		return fmt.Errorf("ATX must be signed before initialization")
```

Comment:
this comment makes less sense to me than the one one.
`Initialize()` is called on the gossip handler path as well as miner generating atx. the msg implied it's only on the miner path.
same for ballot/propsal `Initialize()`